packages/plugins/graphql/server/src/services/builders/mutations/single-type.ts

```diff
@@ -28,56 +26,44 @@ export default ({ strapi }: Context) => {
     const { uid } = contentType;
 
     const updateMutationName = getUpdateMutationTypeName(contentType);
-    const responseTypeName = getEntityResponseName(contentType);
+    const typeName = getTypeName(contentType);
 
     t.field(updateMutationName, {
-      type: responseTypeName,
+      type: typeName,
+
+      extensions: {
+        strapi: {
+          contentType,
+        },
+      },
 
       args: {
         // Update payload
+        status: args.PublicationStatusArg,
         data: nonNull(getContentTypeInputName(contentType)),
       },
 
       async resolve(parent, args, context) {
         const { auth } = context.state;
-        const transformedArgs = transformArgs(args, { contentType });
 
         // Sanitize input data
-        const sanitizedInputData = await sanitize.contentAPI.input(
-          transformedArgs.data,
-          contentType,
-          { auth }
-        );
-
-        Object.assign(transformedArgs, { data: sanitizedInputData });
-
-        const { create, update } = getService('builders')
-          .get('content-api')
-          .buildMutationsResolvers({ contentType });
-
-        // For single types, the validation and sanitization of args is done here instead of being
-        // delegated to the query builders since we're calling the entity service directly
-
-        await validate.contentAPI.query(omit(['data', 'files'], transformedArgs), contentType, {
+        const sanitizedInputData = await sanitize.contentAPI.input(args.data, contentType, {
           auth,
         });
 
-        const sanitizedQuery = await sanitize.contentAPI.query(
-          omit(['data', 'files'], transformedArgs),
-          contentType,
-          {
-            auth,
-          }
-        );
-
-        const entity = (await strapi.entityService!.findMany(uid, sanitizedQuery)) as any;
+        const document = await strapi.db?.query(uid).findOne();
```

Comment:
Why are we using strapi.query here? To avoid lifecycles being triggered?

Reply:
nope because we need a low level query to find the documentId if it exists :) the doc service will always filter on a status or a locale